src/core/vector/qgsvectorlayercache.cpp

```diff
@@ -270,7 +278,13 @@ void QgsVectorLayerCache::onJoinAttributeValueChanged( QgsFeatureId fid, int fie
 void QgsVectorLayerCache::featureDeleted( QgsFeatureId fid )
 {
   mCache.remove( fid );
-  mCacheOrderedKeys.removeOne( fid );
+
+  if ( auto it = mCacheUnorderedKeys.find( fid ); it != mCacheUnorderedKeys.end() )
```

Comment:
C++ 17 at play here? Nice: first time I see that.

Reply:
Yep ... Pretty elegant approach, I'm glad we've got this now!